model_compression_toolkit/core/keras/reader/node_builder.py

```diff
@@ -129,3 +138,25 @@ def __is_functional_inputs_a_list(op_call_args: Any) -> bool:
             inputs_as_list = inputs_as_list and isinstance(arg, KerasTensor)
         return inputs_as_list
     return False
+
+
+def __get_node_distance_fn(layer_class: type, layer_config: Dict[str, Any]) -> Callable:
+    """
+    A mapping between layers' types and a distance function for computing the distance between
+    two tensors (for loss computation purposes). Returns a specific function if node of specific types is
+    given, or a default (normalized MSE) function otherwise.
+    Args:
+        layer_class: Class path of the layer the node represents.
+        layer_config: Framework attributes the layer had which the node holds.
+    Returns: A distance function between two tensors.
+    """
+
+    if layer_class == Activation:
+        node_type_name = layer_config[ACTIVATION]
+        if node_type_name == SOFTMAX or node_type_name == SIGMOID:
+            return compute_kl_divergence
+    elif layer_class == tf.nn.softmax or layer_class == tf.nn.sigmoid:
+        return compute_kl_divergence
+    elif layer_class == Dense:
+        return compute_cs
+    return lambda x, y: compute_mse(x, y, norm=True, norm_eps=1e-8)
```

Comment:
Maybe we should move it to fw implementation and create a function to get interest-points (in fw implementation as well). The get_distance_fn can then use the interest points function to return the correct distance_fn (or None in case it's not an interest point).